crates/indexer/src/models/coin_models/coin_activities.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use super::{
+    coin_balances::{CoinBalance, CurrentCoinBalance},
+    coin_infos::{CoinInfo, CoinSupplyLookup},
+    coin_utils::{CoinEvent, EventGuidResource},
+};
+use crate::{schema::coin_activities, util::truncate_str};
+use aptos_api_types::{
+    Event as APIEvent, Transaction as APITransaction, TransactionInfo as APITransactionInfo,
+    TransactionPayload, UserTransactionRequest, WriteSetChange as APIWriteSetChange,
+};
+use aptos_types::APTOS_COIN_TYPE;
+use bigdecimal::BigDecimal;
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+const BURN_GAS_EVENT: &str = "0x1::aptos_coin::GasBurnEvent";
```

Comment:
There's no event for this.  Are you making up one for your storage?

Reply:
yes